packages/application-generic/src/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -241,14 +242,14 @@ export class TriggerEvent {
               channelType
             );
             if (provider) {
-              providers.set(channelType, provider);
+              providers[channelType] = provider;
             }
           }
         }
       }
     }
 
-    return providers;
+    return providers as Record<ChannelTypeEnum, ProvidersIdEnum>;
```

Comment:
this is a bad thing, you say it's partial and then cast it to the record at the end... please define it as record